pkg/interactor/list_data.go

```diff
@@ -10,94 +10,55 @@ import (
 
 // List provides all the thing's data owned by an user
 func (d *DataInteractor) List(token string, query *entities.Query) ([]entities.Data, error) {
-	data, err := d.getDevicesData(token, query)
+	things, err := d.things.List(token)
 	if err != nil {
-		return []entities.Data{}, err
+		return nil, fmt.Errorf("error getting list of things: %w", err)
```

Comment:
This error should still describe an authorization failure.

Reply:
In this case I suppose that the authorization error will come from `List` method. What do you think?

Reply:
Is this happening?